doc/builders/special/mkshellwrapper.section.md

```diff
@@ -0,0 +1,64 @@
+# pkgs.mkShellWrapper {#sec-pkgs-mkShellWrapper}
+
+`pkgs.mkShellWrapper` is a function that generates a wrapper script that
+receives the command to be run as arguments and then run it inside a special
+environment created with [`mkShell`](#sec-pkgs-mkShell).
+
+It's basically like the `--run` flag of `nix-shell`, but using Nix expressions.
+
+## Usage {#sec-pkgs-mkShellWrapper-usage}
+
+```nix
+{ pkgs ? import <nixpkgs> { } }:
+pkgs.mkShellWrapper {
```

Comment:
I think there should be a better name for this, when I first saw this PR I thought it was a function to wrap shell's like bash and zsh.


Maybe `runInMkShell`

There's `runInLinuxImage` and `runInLinuxVM` in `vmTools`

Reply:
Well. I suggested that actually because the name is like `mkShell`. I also added the documentation for it right after `mkShell` because of SEO, discoverability.

I already did this kind of SEO many times like [when I added a section in Hugo docs](https://github.com/gohugoio/hugoDocs/pull/1471) and opened issues while I was troubleshooting stuff and closed them right after with the solution I found so that would get better SEO and reach more people.

Reply:
changed to runInMkShell